src/beman/execution26/CMakeLists.txt

```diff
@@ -3,34 +3,114 @@
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/schedule.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/scheduler.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/schedule_result_t.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sched_attrs.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sched_env.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sender.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sender_decompose.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sender_for.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/sender_in.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/set_error.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/set_stopped.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/set_value.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/simple_allocator.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/start.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/state_type.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/stoppable_source.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/stoppable_token.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/stop_callback_for_t.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/stop_source.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/transform_sender.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/unstoppable_token.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/valid_completion_for.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/valid_completion_signatures.hpp
+    ${PROJECT_SOURCE_DIR}/include/beman/execution26/detail/valid_specialization.hpp
+)
+get_property(DETAIL_HEADER_FILES TARGET ${TARGET_LIBRARY} PROPERTY HEADER_SET_${TARGET_LIBRARY}_detail_headers)
+source_group("Header Files\\detail" FILES ${DETAIL_HEADER_FILES})
 
 include(GNUInstallDirs)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-target_include_directories(
-  "${TARGET_LIBRARY}"
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../../include>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_LOWER_PROJECT_NAME}>
-)
-
-install(
-  TARGETS "${TARGET_LIBRARY}"
-  EXPORT ${TARGETS_EXPORT_NAME}1
-  DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
 string(TOLOWER ${CMAKE_PROJECT_NAME} CMAKE_LOWER_PROJECT_NAME)
 
 install(
-  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_LOWER_PROJECT_NAME}
-  FILES_MATCHING
-  PATTERN "*.hpp")
+  TARGETS ${TARGET_LIBRARY}
+  EXPORT ${TARGETS_EXPORT_NAME}1
+  ARCHIVE DESTINATION lib/$<CONFIG>
+  FILE_SET ${TARGET_LIBRARY}_public_headers
+  FILE_SET ${TARGET_LIBRARY}_detail_headers
+)
+target_include_directories(${TARGET_LIBRARY} PUBLIC $<INSTALL_INTERFACE:include>)
```

Comment:
I tried to match the install routine from the [Example project](https://github.com/beman-project/Example/tree/main) here.

The only change I made to that is to add a `$<CONFIG>` suffix to the library install directory. This is to allow better support for multi-config generators. I opened[ an issue on the example repo](https://github.com/beman-project/Example/issues/18) to discuss the issue.